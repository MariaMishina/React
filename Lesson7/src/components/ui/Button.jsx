import PropTypes from 'prop-types'
import styles from './Button.module.css'

export function Button(props) {
  return (
    <>
      <button className={styles.myBut} {...props}>{props.children}</button>
    </>
  )
}

Button.propTypes = {
  type: PropTypes.string
}