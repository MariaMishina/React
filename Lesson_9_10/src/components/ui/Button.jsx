import PropTypes from 'prop-types'
import styles from './Button.module.css'

export function Button(props) {
  return (
    <>
      <button style={{color: 'black'}} className={styles.myBut} {...props} onClick={props.click}>{props.children}</button>
    </>
  )
}

Button.propTypes = {
  type: PropTypes.string
}