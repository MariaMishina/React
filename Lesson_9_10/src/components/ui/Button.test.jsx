import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
import {Button} from './Button'

describe('testing Button', () => {
  it('render component Button', () => {
    render(<Button>Open</Button>)
  })

  it('render with snapshot', () => {
   const {asFragment} = render(<Button>Open</Button>)
   expect(asFragment()).toMatchSnapshot()
  })

  it('render component with text name button', () => {
    render(<Button>Open</Button>)
    expect(screen.getByText(/Open/)).toBeInTheDocument()
  })

  it('render multiply component', () => {
    render(
      <>
        <Button>Click</Button>
        <Button>Clack</Button>
      </>
    )
    expect(screen.queryAllByRole('button').length).toBe(2)
  })

  it('button is disable', () => {
    render(<Button disabled>Open</Button>)
    expect(screen.getByText(/Open/)).toBeDisabled()
  })

  it('button have style color black', () => {
    render(<Button>Open</Button>)
    expect(screen.getByText(/Open/)).toHaveStyle({color: 'black'})
  })

  it('button click wwith userEvent', async () => {
    const mockHandler = jest.fn()

    render(<Button click={mockHandler}>Open</Button>)
    await userEvent.click(screen.getByText(/Open/))
    expect(mockHandler).toHaveBeenCalledTimes(1)
  })

  it('test example', async () => {
    const onChange = jest.fn()
    render(<input type="checkbox" onChange={onChange} />)

    const checkbox = screen.getByRole('checkbox')
    await userEvent.dblClick(checkbox)
    expect(onChange).toHaveBeenCalledTimes(2)
    expect(checkbox).not.toBeChecked()
  })
  
  
})