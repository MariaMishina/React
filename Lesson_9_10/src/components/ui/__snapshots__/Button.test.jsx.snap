// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="myBut"
    style="color: black;"
  >
    Open
  </button>
</DocumentFragment>
`;
